test/e2e/path_based_test.go

```diff
@@ -145,6 +146,118 @@ func TestPathBasedRouting(t *testing.T) {
+	imageTag, tagPres := os.LookupEnv("dockerVersion")
+
+	if !repoPres {
+		imageRepo = "ml-docker-db-dev-tierpoint.bed-artifactory.bedford.progress.com/marklogic/marklogic-server-centos"
+		t.Logf("No imageRepo variable present, setting to default value: " + imageRepo)
+	}
+
+	if !tagPres {
+		imageTag = "11.0.nightly-centos-1.0.2"
+		t.Logf("No imageTag variable present, setting to default value: " + imageTag)
+	}
+
+	namespaceName := "ml-" + strings.ToLower(random.UniqueId())
+	kubectlOptions := k8s.NewKubectlOptions("", "", namespaceName)
+
+	// Setup the args for helm install using custom values.yaml file
+	options := &helm.Options{
+		ValuesFiles:    []string{"../test_data/values/tls_pbr_appser_values.yaml"},
+		KubectlOptions: k8s.NewKubectlOptions("", "", namespaceName),
+	}
+
+	t.Logf("====Creating namespace: " + namespaceName)
+	k8s.CreateNamespace(t, kubectlOptions, namespaceName)
+
+	defer t.Logf("====Deleting namespace: " + namespaceName)
+	defer k8s.DeleteNamespace(t, kubectlOptions, namespaceName)
+
+	t.Logf("====Installing Helm Chart")
+	releaseName := "test-path"
+	helm.Install(t, options, helmChartPath, releaseName)
+
+	podName := releaseName + "-1"
+	svcName := releaseName + "-haproxy"
+
+	// wait until the pod is in Ready status
+	k8s.WaitUntilPodAvailable(t, kubectlOptions, podName, 15, 20*time.Second)
+
+	tunnel := k8s.NewTunnel(
+		kubectlOptions, k8s.ResourceTypeService, svcName, 8080, 80)
+	defer tunnel.Close()
+	tunnel.ForwardPort(t)
+
+	client := req.C().
+		SetCommonBasicAuth(username, password).
+		SetCommonRetryCount(10).
+		SetCommonRetryFixedInterval(10 * time.Second)
+
+	endpoint := "http://localhost:8080/manage/manage/v2/servers?group-id=Default&server-type=http&format=json"
+	fmt.Println(endpoint)
+	testServerReq, err := os.ReadFile("../test_data/path_based_test_data/test-server.json")
+	if err != nil {
+		fmt.Print(err)
+	}
+
+	//create new app server: test-server
+	resp, err := client.R().
+		SetHeader("Content-type", "application/json").
+		SetBodyJsonString(string(testServerReq)).
+		AddRetryCondition(func(resp *req.Response, err error) bool {
+			if err != nil {
+				t.Logf("error: %s", err.Error())
+			}
+			t.Logf("StatusCode: %d", resp.GetStatusCode())
+			if resp.GetStatusCode() != 201 {
+				t.Log("Waiting for MarkLogic cluster to be ready")
+			}
+			return resp.GetStatusCode() != 201
+		}).
+		Post(endpoint)
+	if err != nil {
+		t.Fatalf(err.Error())
+	}
+	defer resp.Body.Close()
+
+	//test the additional app servers path
+	path := "test"
+	endpoint = fmt.Sprintf("http://localhost:8080/%s", path)
+	t.Logf("Verifying path based routing using %s", endpoint)
+	resp, err = client.R().
+		AddRetryCondition(func(resp *req.Response, err error) bool {
+			if err != nil {
+				t.Logf("error: %s", err.Error())
+			}
+			t.Logf("StatusCode: %d", resp.GetStatusCode())
+			if resp.GetStatusCode() != 500 {
+				t.Log("Waiting for MarkLogic cluster to be ready")
+			}
+			return resp.GetStatusCode() != 500
```

Comment:
We should check for a specific response instead of a generic failure.

Reply:
As there is no application running on that app server it would return 500. Added check for response as well.